### DISCUSSION TITLE
Error while querying single element with a custom field validation

### DISCUSSION THREAD

**Discussion by jesusmdy** [strapi/strapi#7529] [Legacy Topics]
Hi, I am developing a web application with Vue, Vue-Apollo and Strapi with GraphQL.
But I've been trying to query `category`'s collection wanting to get a single `item` using the field `slug` as identifier.
```
query getSingleCategory($categorySlug: String!) {
  categories(slug: $categorySlug) {
    name
    slug
  }
}
```
### With the variable
```
{
  "categorySlug": "news-and-advirsement"
}
```
### And i get this error
```
{
  "error": {
    "errors": [
      {
        "message": "Unknown argument \"name\" on field \"categories\" of type \"Query\".",
        "locations": [
          {
            "line": 2,
            "column": 14
          }
        ],
        "extensions": {
          "code": "GRAPHQL_VALIDATION_FAILED",
          "exception": {
            "stacktrace": [
              "GraphQLError: Unknown argument \"name\" on field \"categories\" of type \"Query\".",
              "    at Object.Argument (C:\\www\\coindex-api\\node_modules\\graphql\\validation\\rules\\KnownArgumentNames.js:60:29)",
              "    at Object.enter (C:\\www\\coindex-api\\node_modules\\graphql\\language\\visitor.js:324:29)",
              "    at Object.enter (C:\\www\\coindex-api\\node_modules\\graphql\\language\\visitor.js:375:25)",
              "    at visit (C:\\www\\coindex-api\\node_modules\\graphql\\language\\visitor.js:242:26)",
              "    at Object.validate (C:\\www\\coindex-api\\node_modules\\graphql\\validation\\validate.js:73:24)",
              "    at validate (C:\\www\\coindex-api\\node_modules\\apollo-server-core\\dist\\requestPipeline.js:221:34)",
              "    at Object.<anonymous> (C:\\www\\coindex-api\\node_modules\\apollo-server-core\\dist\\requestPipeline.js:118:42)",
              "    at Generator.next (<anonymous>)",
              "    at fulfilled (C:\\www\\coindex-api\\node_modules\\apollo-server-core\\dist\\requestPipeline.js:5:58)",
              "    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
            ]
          }
        }
      }
    ]
  }
}
```
![image](https://user-images.githubusercontent.com/26631510/90715915-965fc380-e279-11ea-8289-5a8e3b7ec627.png)

### And
This is my collection type
![image](https://user-images.githubusercontent.com/26631510/90716063-02422c00-e27a-11ea-8a81-353efed1eeb6.png)

### But...
If i run the query without field validation it seems to work properly
![image](https://user-images.githubusercontent.com/26631510/90715963-b8594600-e279-11ea-8f46-71b6cc884a72.png)

### So
I really were testing different queries and doing researching about `single query with another field verification` and it still failing.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by jesusmdy** [reply]
Thanks, i'll take a look.

**Comment by derrickmehaffy** [comment]
@jesusmdy you are using the plural form of the model and in this case to filter you need to use `categories ( where: { slug: "news-and-advirsement" } )` However if you plan to use this with a variable you need to send the entire `$where` structure as a variable like this:

```
query getSingleCategory($whereSlug: JSON!) {
  categories( where: $whereSlug ) {
    name
    slug
  }
}
```

Vars
```
{
  "whereSlug": {
    "slug": "news-and-advirsement"
  }
}
```

**Reply by jesusmdy** [reply]
Seems to work! Thanks!

**Reply by derrickmehaffy** [reply]
No problem! :)
